src/Client/Updater.php

```diff
@@ -398,31 +420,36 @@ private function updateRoot(RootMetadata &$rootData)
         $originalRootData = $rootData;
         // *TUF-SPEC-v1.0.9 Section 5.1.2
         $nextVersion = $rootData->getVersion() + 1;
-        while ($nextRootContents = $this->repoFileFetcher->fetchFile("$nextVersion.root.json", static::MAXIMUM_DOWNLOAD_BYTES)) {
-            $rootsDownloaded++;
-            if ($rootsDownloaded > static::MAX_ROOT_DOWNLOADS) {
-                throw new \Exception("The maximum number root files have already been dowloaded:" . static::MAX_ROOT_DOWNLOADS);
+        try {
+            while ($nextRootContents = $this->repoFileFetcher->fetchFile("$nextVersion.root.json", static::MAXIMUM_DOWNLOAD_BYTES)) {
```

Comment:
This nesting is now need because TestRepo was not implementing `\Tuf\Client\RepoFileFetcherInterface::fetchFile` correctly.